lib/UserResolver.php

```diff
@@ -24,15 +24,25 @@ public function __construct(IUserManager $userManager) {
 	 * @throws NoUserFoundException
 	 */
 	public function findExistingUserId(string $rawUidCandidate, bool $force = false): string {
-		if ($force) {
-			$this->ensureUser($rawUidCandidate);
+		if ($force)
+		{
+			$usersFound=$this->ensureUser($rawUidCandidate);
+			if(count($usersFound) != 1)
+			{
+				throw new NoUserFoundException('User' . $rawUidCandidate . ' not valid or not found');
```

Comment:
This will work on maybe 99% of the cases, but not where the search attributes are configured in a way that would not reliably find the user.

Should not throw but continue with previous behaviour.